### PR TITLE
cpu/msp430: fix spi clk config for msp430f1xxx

### PR DESCRIPTION
**Pull request by PeterKietzmann**

### CODE REVIEW

File: cpu/msp430fxyz/periph/spi.c [hunk 1]
```diff
@@ -52,18 +52,21 @@ int spi_init_master(spi_t dev, spi_conf_t conf, spi_speed_t speed)
     SPI_DEV->TCTL = (USART_TCTL_SSEL_SMCLK | USART_TCTL_STC);
     /* set polarity and phase */
     switch (conf) {
+        case SPI_CONF_FIRST_RISING:
+            SPI_DEV->TCTL |= (USART_TCTL_CKPH & ~(USART_TCTL_CKPL));
```
Comment (haukepetersen) [inline cpu/msp430fxyz/periph/spi.c]: the second part is without effect, simply `SPI_DEV->TCTL |= (USART_TCTL_CKPH);` is enoug?!

File: cpu/msp430fxyz/periph/spi.c [hunk 2]
```diff
@@ -52,18 +52,21 @@ int spi_init_master(spi_t dev, spi_conf_t conf, spi_speed_t speed)
...
+            break;
         case SPI_CONF_SECOND_RISING:
-            SPI_DEV->TCTL |= USART_TCTL_CKPH;
+            SPI_DEV->TCTL |= (~(USART_TCTL_CKPH) & ~(USART_TCTL_CKPL));
```
Comment (haukepetersen) [inline cpu/msp430fxyz/periph/spi.c]: line without effect...

File: cpu/msp430fxyz/periph/spi.c [hunk 3]
```diff
@@ -52,18 +52,21 @@ int spi_init_master(spi_t dev, spi_conf_t conf, spi_speed_t speed)
...
             break;
         case SPI_CONF_FIRST_FALLING:
-            SPI_DEV->TCTL |= SPI_CONF_FIRST_FALLING;
+            SPI_DEV->TCTL |= (USART_TCTL_CKPH & USART_TCTL_CKPL);
             break;
         case SPI_CONF_SECOND_FALLING:
-            SPI_DEV->TCTL |= (USART_TCTL_CKPH | SPI_CONF_FIRST_FALLING);
+            SPI_DEV->TCTL |= (~(USART_TCTL_CKPH) & USART_TCTL_CKPL);
```
Comment (haukepetersen) [inline cpu/msp430fxyz/periph/spi.c]: drop first part here, too

**Comment by PeterKietzmann** [comment]
@haukepetersen thanks for the review. I was inspired by the initialization code for SPI devices that use `SPI_USE_USCI` and found it great in terms of readability. However, I removed the stuff you commented for both MCU variants. Will give it a try this afternoon (just to be on the save side).

**Comment by PeterKietzmann** [comment]
All right this works. May I squash and hit the RfCIb label?

**Comment by kYc0o** [comment]
It didn't break the z1? It was the only one working before this PR.

**Comment by PeterKietzmann** [comment]
Nope because the z1 has an msp43f2xxx MCU which uses the `USCI` SPI and thus the initialization in line 110

**Comment by kYc0o** [comment]
Oh yeah, I didn't see the small "1" before xyz. Great!

**Comment by PeterKietzmann** [comment]
This is a bugfix which should definitely go into the release

File: cpu/msp430fxyz/periph/spi.c [hunk 4]
```diff
@@ -122,20 +125,20 @@ int spi_init_master(spi_t dev, spi_conf_t conf, spi_speed_t speed)
     /* set polarity and phase */
     switch (conf) {
         case SPI_CONF_FIRST_RISING:
-            SPI_DEV->CTL0 |= (USCI_SPI_CTL0_CKPH & ~(USCI_SPI_CTL0_CKPL));
+            SPI_DEV->CTL0 |= USCI_SPI_CTL0_CKPH;
             break;
         case SPI_CONF_SECOND_RISING:
-            SPI_DEV->CTL0 |= (~(USCI_SPI_CTL0_CKPH) & ~(USCI_SPI_CTL0_CKPL));
+            /* nothong to be done here */
```
Comment (miri64) [inline cpu/msp430fxyz/periph/spi.c]: *Nothing

**Comment by PeterKietzmann** [comment]
Addressed latest comment, squashed into two reasonable(?) commits and hit the RfCIb button

**Comment by haukepetersen** [comment]
ACK once @miri64s comment is fixed.

**Comment by PeterKietzmann** [comment]
It is already addressed. Probably the above view hasn't been updated cause I squashed immediately.

**Comment by miri64** [comment]
> It is already addressed. Probably the above view hasn't been updated cause I squashed immediately.

No, that's just GitHub's new interface...

**Comment by miri64** [comment]
Ooops, sorry hit the wrong checkbox.

**Comment by miri64** [comment]
@kaspar030 Murdock is being weird again... according to the tally at https://ci.riot-labs.de/ the PR build was successful, but the build status here indicates that the build only has been started.... Regardless, I'm merging now.

**Pull request merged**